controller/repliesController.js

```diff
@@ -34,6 +34,68 @@ const getCommentReplies = async (req, res, next) => {
   }
 };
 
+const createReply = async (req, res, next) => {
+  try {
+    //validation should be done via middleware
+    //ownerId in body also needs to be validated
+
+    const { ownerId, content } = req.body;
+    const { commentId } = req.params;
+
+    if (!ObjectId.isValid(commentId)) {
+      next(new CustomError(404, "invalid ID"));
+      return;
+    }
+
+    if (!ownerId || !content) {
+      next(new CustomError(422, `Enter the required fields`));
+      return;
+    }
+    const reply = new Replies({
+      content,
+      ownerId,
+      commentId,
+    });
+
+    const savedReply = await reply.save();
+    const parentComment = await Comments.findByIdAndUpdate(
+      commentId,
+      {
+        $push: {
+          replies: savedReply._id,
+        },
+      },
+      {
+        new: true,
+      }
+    );
+    if (!parentComment) {
+      next(
+        new CustomError(
+          404,
+          `Comment with the ID ${commentId} does not exist or has been deleted`
+        )
+      );
+      return;
+    }
+    const data = {
+      replyId: savedReply._id,
+      commentId: savedReply.commentId,
+      content: savedReply.content,
+      ownerId: savedReply.ownerId,
+      upVotes: savedReply.upVotes,
+      downVotes: savedReply.downVotes,
+      flags: savedReply.flags,
+    };
+
+    responseHandler(res, 201, data, "Reply added successfully");
+    return;
+  } catch (error) {
+    next(error);
```

Comment:
This should be using the CustomError class. Here's an example
```suggestion
    next(new CustomError(500, "Something went wrong, please try again later", error));
```